geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerFeatureSource.scala

```diff
@@ -65,7 +67,13 @@ class KafkaConsumerFeatureSource(entry: ContentEntry,
     }
   }
 
-  val features = Maps.newConcurrentMap[String, FeatureHolder]()
+  val cb = CacheBuilder.newBuilder()
+
+  if (expiry) {
+    cb.expireAfterWrite(expirationPeriod, TimeUnit.MILLISECONDS)
```

Comment:
You should add a RemovalListener here that will remove the feature from the QuadTree when it gets evicted from the cache.